### PR TITLE
Pass parameter to get feed items()

### PR DESCRIPTION
**Pull request by leedriscoll**
Added ability to pass a filter with items in the config.

You can now do this
```php
return [
    'feeds' => [
        [
            'items' => ['App\NewsItem@getFeedItemsForCategory', 'news'],
            ...
```
As well as this:
```php
return [
    'feeds' => [
        [
            'items' => 'App\NewsItem@getFeedItemsForCategory',
            ...
```

### CODE REVIEW

**Comment by freekmurze** [comment]
Thanks, will review it soon. Don't mind that failing styleci check. That'll be fixed automatically.

File: src/Feed.php [hunk 1]
```diff
@@ -26,15 +26,33 @@ public function getFeedResponse()
...
 
-        $items = app($class)->{$method}();
+        $items = app($class)->{$method}($this->getFeedConfigurationItemsFilter());
 
         $meta = ['id' => url($this->feedConfiguration['url']), 'link' => url($this->feedConfiguration['url']), 'title' => $this->feedConfiguration['title'], 'updated' => $this->getLastUpdatedDate($items)];
 
         return view('laravel-feed::feed', compact('meta', 'items'))->render();
     }
 
+    protected function getFeedConfigurationItems()
```
Comment (freekmurze) [inline src/Feed.php]: Rename this to `getFeedArguments`

File: src/Feed.php [hunk 2]
```diff
@@ -26,15 +26,33 @@ public function getFeedResponse()
...
+    {
+        if (is_array($this->feedConfiguration['items'])) {
+            return $this->feedConfiguration['items'][0];
+        }
+
+        return $this->feedConfiguration['items'];
+    }
+
+    protected function getFeedConfigurationItemsFilter()
```
Comment (freekmurze) [inline src/Feed.php]: Rename this to `getFeedArguments`

File: src/Feed.php (hunk 1 above)
Comment (freekmurze) [inline src/Feed.php]: Rename this to `getFeedMethod` (or something better to reflect that it's actually the classname and the method).

File: tests/DummyRepository.php [hunk 3]
```diff
@@ -14,4 +14,21 @@ public function getAll()
             new DummyItem(),
         ]);
     }
+
+    public function getAllWithFilter($filter = '')
```
Comment (freekmurze) [inline tests/DummyRepository.php]: Rename this to `getAllWithArguments`

File: src/Feed.php [hunk 4]
```diff
@@ -26,15 +26,33 @@ public function getFeedResponse()
...
+    {
+        if (is_array($this->feedConfiguration['items'])) {
```
Comment (freekmurze) [inline src/Feed.php]: Using a ternary operator may improve readability here.

File: src/Feed.php [hunk 5]
```diff
@@ -26,15 +26,33 @@ public function getFeedResponse()
...
+        $items = app($class)->{$method}($this->getFeedConfigurationItemsFilter());
 
         $meta = ['id' => url($this->feedConfiguration['url']), 'link' => url($this->feedConfiguration['url']), 'title' => $this->feedConfiguration['title'], 'updated' => $this->getLastUpdatedDate($items)];
 
         return view('laravel-feed::feed', compact('meta', 'items'))->render();
     }
 
+    protected function getFeedConfigurationItems()
+    {
+        if (is_array($this->feedConfiguration['items'])) {
```
Comment (freekmurze) [inline src/Feed.php]: Using a ternary operator may improve readability here.

**Comment by leedriscoll** [comment]
Thanks for the feedback @freekmurze. 
I've made all the changes you've suggested 👍

**Comment by freekmurze** [comment]
Thank you for this, I'll fix some small nitpicks myself and tag a new release.

**Pull request merged**